local mount, profiles and new docs

MdreW opened the pull request:
This is an example for a docker compose without static volumes, has only example scope

* static mount are removed
* added directory Docker-compose/satosa for satosa container application
* added directory Docker-compose/djangosaml2_sp/* for django container application
* added directory Docker-compose/mongo/db/* for mongo container db
* updated .gitignore for exclude production data
* removed rm-persistent-volumes.sh script
* Updated run-docker-compose.sh
* * Create if no exists required folder
* * Added `-f` option for clean old dp/application data and reinitialize the sistem
* * Added `-h` option for print help
```
$ run-docker-compose.sh

### run-docker-compose.sh ###

initialize and start Satosa-Saml2Spid

Option
-f Force clean and reinitialize data for Satosa, MongoDB and Djangosaml2_SP
-h Print this help
```

peppelinux commented on Docker-compose/docker-compose.yml:
```diff
@@ -9,8 +9,7 @@ services:
       MONGO_INITDB_ROOT_PASSWORD: "${MONGO_DBPASSWORD:-thatpassword}"
     volumes:
-      - mongodata:/data/db
-      - /usr/share/zoneinfo/Europe/Rome:/etc/localtime:ro
```
This Is needed for the local timezone within the container

MdreW replied:
I don't think it's necessary, this work is already execute in line 23 of Dockerfile
https://github.com/italia/Satosa-Saml2Spid/blob/b26f78fb330e8c09d000cc569f713c9201ab9700/Dockerfile#L23

MdreW replied:
Hi @peppelinux , Apologize, I didn't realize you were referring to MongoDB.

MongoDB save the datetime object in UTC format (without explicit rule), for this container  the timezone is not very important. Anyway, MongoDB official image, support the TZ environment  to set the timezone:

```bash
[ranaldi@andrea-ranaldi-p ~]$ docker exec -tie TZ=Europe/Rome mongo bash
root@b8714ec55ff8:/# date
Mon Aug 26 15:59:00 CEST 2024
root@b8714ec55ff8:/# exit
exit
[ranaldi@andrea-ranaldi-p ~]$ docker exec -tie TZ=Australia/Melbourne mongo bash
root@b8714ec55ff8:/# date
Mon Aug 26 23:59:13 AEST 2024
root@b8714ec55ff8:/# 
```
I add the TZ environment in the docker-compose file

peppelinux replied:
I don't want that the docker image uses a static timezone, but it would be better to have it configurable through local mount points

for all the rest in this PR I fully approve the approach that really simplify all the things!

thx @MdreW

peppelinux commented on Docker-compose/docker-compose.yml:
```diff
@@ -110,8 +108,7 @@ services:
     volumes:
-       - /usr/share/zoneinfo/Europe/Rome:/etc/localtime:ro
-       - ../docker-example:/satosa_proxy:rw
+       - ./satosa:/satosa_proxy:rw
```
satosa -> project?

I agree that `docker-example` is too much related to the demo purposes, while satosa is too much generic and might confuse the software/application name with the local configuration, I suppose that `project` or any other similar alternative might better explain the purpose of that folder

MdreW commented:
Added TZ in docker compose, update satosa Dockerfile to support this.
a little optimization of Dockerfile RUN

MdreW commented:
I have added profiles demo to start all services and dataentry for add mongo-express service.
Nginx, satosa and mongo are every started for production.

I have update update `run-docker-compose.sh` to start `demo` profile as default,  `dataentry` and `production`  profiles are optional. I have added also an option to skip docker image update and compile `-s`.

```
### run-docker-compose.sh ###

initialize check update and start Satosa-Saml2Spid

Option
-f Force clean and reinitialize data for Satosa, MongoDB and Djangosaml2_SP
-h Print this help
-p Set production profile: start satosa, nginx, mongo
-s Skip docker image update
-d Set data entry profile: start satosa, nginx, mongo, mongo-express
   if isn't set -d or -p defatult demo profile is started
   default demo profile start: satosa, nginx, mongo, mongo-express, django-sp, spid-saml-check
```

At last I have update `./stop-docker-compose.sh` to stop all profiles and remove orphan containers, but we can add an option to stop also only specific profiles.

@Zicchio you can try this on windows?

peppelinux commented on Docker-compose/docker-compose.yml:
```diff
@@ -1,55 +1,57 @@
+    depends_on:
+      - satosa-mongo
     environment:
       ME_CONFIG_BASICAUTH_USERNAME: satosauser
```
Wouldnt be mettere to have them in ENV?

MdreW replied:
For instances with only spid-saml backend? I can try :+1:

MdreW replied:
I have tried and I can't use an Environment to enable mongo dependence on compose because `depends_on` don't accept "null" as value:
`service "satosa-saml2spid" depends on undefined service "": invalid` 

I can propose a new docker compose profile for MongoDB? The profiles can be:

**default (without profile):**
* satosa-sa,l2spid (without mongo dependancy)
* satosa-nginx

**db:**
* *default
*  satosa-mongo

**dataentry:**
* *default
* *db
* satosa-mongoexpress

**dev:**
* *default
* django-sp
* spid-samlcheck

**demo:**
* *default**
* *db
* *dataentry
*  *dev

If I use an external MongoDB, is sufficient don't start the `db` profile (only default) or use `dev` profile for a full system without mongo.
What do you think?

peppelinux commented on Docker-compose/docker-compose.yml:
```diff
@@ -110,8 +113,8 @@ services:
     volumes:
-       - /usr/share/zoneinfo/Europe/Rome:/etc/localtime:ro
-       - ../docker-example:/satosa_proxy:rw
+      - ./satosa:/satosa_proxy:rw
```
```suggestion
      - ./satosa-project:/satosa_proxy:rw
```

MdreW commented:
Fix #156

MdreW commented:
Migrate to #160
